src/oceans/components/common/Guide.jsx

```diff
@@ -1,29 +1,82 @@
-import React from 'react'
-import Radium from "radium";
-import Typist from "react-typist";
-
-import {getState, setState} from "@ml/oceans/state";
-import guide from "@ml/oceans/models/guide";
-import soundLibrary from "@ml/oceans/models/soundLibrary";
-import styles from "@ml/oceans/styles";
-import colors from "@ml/oceans/styles/colors";
-import I18n from "@ml/oceans/i18n";
-import {Button} from "@ml/oceans/components/common";
-import arrowDownImage from "@public/images/arrow-down.png";
+import React from 'react';
+import Radium from 'radium';
+import Typist from 'react-typist';
+
+import {getState, setState} from '@ml/oceans/state';
+import guide from '@ml/oceans/models/guide';
+import soundLibrary from '@ml/oceans/models/soundLibrary';
+import styles from '@ml/oceans/styles';
+import colors from '@ml/oceans/styles/colors';
+import I18n from '@ml/oceans/i18n';
+import {Button} from '@ml/oceans/components/common';
+import arrowDownImage from '@public/images/arrow-down.png';
+import {
+  startTextToSpeech,
+  stopTextToSpeech,
+  hasTextToSpeechVoices
+} from '@ml/utils/TextToSpeech';
+
+// A timer used for playing typing sounds.
+let guideTypingTimer = undefined;
```

Comment:
Is this no longer needed as state? I still see it referenced in a couple places (eg, `index.jsx`)

Reply:
Calling `setState` from our `render` function was trigging a (justifiable) warning from React:
```
Render methods should be a pure function of props and state; triggering nested component updates from render is not allowed. If necessary, trigger nested updates in componentDidUpdate.
```

At first I hoisted some variables out of state to avoid this, since changes to them don't need to trigger re-renders, but then I remembered that we had an optional `skipCallback` parameter in `setState` so I used that instead.